rust/catalyst-types/src/uuid/mod.rs

```diff
@@ -32,16 +33,30 @@ pub enum CborUuidError {
     },
 }
 
-/// Decode `CBOR` encoded `UUID`.
-pub(crate) fn decode_cbor_uuid(val: &coset::cbor::Value) -> Result<uuid::Uuid, CborUuidError> {
-    let Some((UUID_CBOR_TAG, coset::cbor::Value::Bytes(bytes))) = val.as_tag() else {
-        return Err(CborUuidError::InvalidCborType);
-    };
-    let uuid = uuid::Uuid::from_bytes(
-        bytes
-            .clone()
-            .try_into()
-            .map_err(|_| CborUuidError::InvalidByteSize)?,
-    );
+/// Decode from `CBOR` into `UUID`
+pub(crate) fn decode_cbor_uuid(
+    d: &mut Decoder<'_>, (): &mut (),
+) -> Result<uuid::Uuid, minicbor::decode::Error> {
```

Comment:
For generality, there are three possible options one would want on a decode:
1. MUST have a TAG
2. MUST not have a TAG
3. Optionally has a TAG.

Suggest adding a `tagged: Option<bool>` as the context.
* None = Optionally has a Tag
* Some(true) = Must have a Tag
* Some(false) = Must NOT have a Tag